core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/LockCache.java

```diff
@@ -95,32 +97,29 @@ private void throwCachedLockException(LockException e) throws LockException
     static final class LockKey
     {
         private final String myDataCenter;
-        private final String myResource;
+        private final String myResourceName;
 
         LockKey(String dataCenter, String resource)
         {
+            Preconditions.checkNotNull(resource);
             myDataCenter = dataCenter;
-            myResource = resource;
+            myResourceName = resource;
         }
 
         @Override
         public boolean equals(Object o)
         {
             if (this == o) return true;
             if (o == null || getClass() != o.getClass()) return false;
-
             LockKey lockKey = (LockKey) o;
-
-            if (myDataCenter != null ? !myDataCenter.equals(lockKey.myDataCenter) : lockKey.myDataCenter != null) return false;
-            return myResource.equals(lockKey.myResource);
+            return Objects.equals(myDataCenter, lockKey.myDataCenter) &&
+                    myResourceName.equals(lockKey.myResourceName);
```

Comment:
Personally I like the "symmetry" by using Objects.equals on both members... Even though you need 1 extra method call on the stack when using it...
return Objects.equals(myDataCenter, that.myDataCenter) &&
                Objects.equals(myResourceName, that.myResourceName);

Reply:
Changed both implementations to use Objects.equals instead